woocommerce/google-listings-and-ads
# Attribute Mapping - CRUD Rules Frontend

### Changes proposed in this Pull Request:



Closes part of #1648.

This PR implements the Create/Read/Update/Delete features in the Attribute Mapping Rules table.

Pagination is also included.

### Screenshots:

https://user-images.githubusercontent.com/5908855/193570814-2f71fdea-6a08-400e-94f2-18a31ab05153.mov


https://user-images.githubusercontent.com/5908855/194066435-f577e5e8-439b-4958-8a5e-4ef84bd1e75d.mov



### Detailed test instructions:


1. GO to Attribute Mapping Tab
2. See that when data is loading a placeholder table appears
3. Click Create a new rule and se how it appears in the table (notice save is disabled if the form is not correct)
4. Save the rule (see Saving... state)
5. Refresh and see how the data appears after reloading
6. Edit the rule and see that Save button is disabled if form is not correct or didn't change
7. After saving, see how it is modified in the table
8. Refresh and see how the data appears correctly after reloading
9. Delete some rule and see the modal 
10. Confirm deletion and see how the rule is deleted
11. Test the pagination by creating more than 10 rules and deleting some of them.

### Additional details:

⚠️  Notice `source_name` (with the nice name for the sources) is not yet implemented. 

Figma: https://www.figma.com/file/fqR0EHi63lWahRcVTKCcba/Google-Listings-%26-Ads-v2.x?node-id=1610%3A156293
